data/model/oci/manifest.py

```diff
@@ -251,6 +273,9 @@ def get_or_create_manifest(
         raise_on_error=raise_on_error,
         retriever=retriever,
     )
+    if created_manifest is not None:
+        reset_child_manifest_expiration(repository_id, created_manifest.manifest)
+    return created_manifest
```

Comment:
I suppose not, but it does help GC run a bit quicker. We do have this hidden behind a config flag for now. 

This will run on the push of a manifest. This will also update every alive temporary tag pointing to every manifest in the manifest list. In certain circumstances this could be large, ie. if it's the first time this code runs and the manifest has been referenced multiple times on the manifest list push.

I'm not sure of another way of doing this except:
- Setting `PUSH_TEMP_TAG_EXPIRATION_SEC` to a lower value
- Setting the expiration of temporary tags of the child manifests on parent tag deletion in the `DELETE /tag` endpoint
- Setting the expiration of temporary tags of the child manifests when the parent tag is GC'd